internal/cmd/gocdk/static/demo/blob.bucket/demo_blob_bucket.go

```diff
@@ -0,0 +1,186 @@
+  </p>
+  <p>
+    It is currently using a blob.Bucket based on the URL "{{ . }}", which
+    can be configured via the environment variable "BLOB_BUCKET_URL".
+  </p>
+  <p>
+    See <a href="https://godoc.org/gocloud.dev#hdr-URLs">here</a> for more
+    information about URLs in Go CDK APIs.
+  </p>
+  <ul>
+    <li><a href="./list">List</a> the contents of the bucket</li>
+    <li><a href="./view">View</a> the contents of a specific blob in the bucket</li>
+    <li><a href="./write">Write</a> a new blob into the bucket</li>
+  </ul>
+</body>
+</html>
+`
+
+func blobBucketBaseHandler(w http.ResponseWriter, req *http.Request) {
+	tmpl := template.Must(template.New("base").Parse(blobBucketBaseTemplate))
+	err := tmpl.Execute(w, bucketURL)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+// Input: []*blob.ListObject.
+const blobBucketListTemplate = `
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="UTF-8">
+  <title>blob.Bucket demo</title>
+</head>
+<body>
+  {{range .}}
+    <div>
+      {{if .IsDir}}
+        <a href="/demo/blob/list?prefix={{ .Key }}">{{ .Key }}</a>
+      {{else}}
+        <a href="/demo/blob/view?key={{ .Key }}">{{ .Key }}</a>
+      {{end}}
+    </div>
+  {{else}}
+    <div>no blobs in bucket</div>
+  {{end}}
+</body>
+</html>
+`
+
+// blobBucketListHandler lists the items in a bucket, possibly under a "prefix"
+// query parameter. Each listed directory is a link to list that directory,
+// and each non-directory is a link to view that file.
+func blobBucketListHandler(w http.ResponseWriter, req *http.Request) {
+	if bucketErr != "" {
+		http.Error(w, bucketErr, http.StatusInternalServerError)
+		return
+	}
+	opts := &blob.ListOptions{
+		Delimiter: "/",
+		Prefix:    req.FormValue("prefix"),
+	}
+	iter := bucket.List(opts)
+	ctx := req.Context()
+	var items []*blob.ListObject
+	for {
+		item, err := iter.Next(ctx)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to iterate to next blob.Bucket key: %v", err), http.StatusInternalServerError)
+			return
+		}
+		items = append(items, item)
+	}
+
+	tmpl := template.Must(template.New("list").Parse(blobBucketListTemplate))
+	err := tmpl.Execute(w, items)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func blobBucketViewHandler(w http.ResponseWriter, req *http.Request) {
+	if bucketErr != "" {
+		http.Error(w, bucketErr, http.StatusInternalServerError)
+		return
+	}
+	// TODO(rvangent): By default, render a form with key name drop-down instead of
+	// erroring out.
+	key := req.FormValue("key")
+	if key == "" {
+		http.Error(w, "\"key\" is a required parameter.", http.StatusBadRequest)
+		return
+	}
+
+	ctx := req.Context()
+	reader, err := bucket.NewReader(ctx, key, nil)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Failed to create Reader: %v\n", err), http.StatusNotFound)
+		return
+	}
+	defer reader.Close()
+	io.Copy(w, reader)
```

Comment:
Also set `Content-Type` and `Content-Length` headers

Reply:
Add a TODO for this. It doesn't play nice with `fileblob`, as none of the blobs on the filesystem have `ContentType`, so it ends up being "application/octet-stream", so the browser downloads them instead of trying to display them.